static/js/ducks/instrument.js

```diff
@@ -0,0 +1,44 @@
+import messageHandler from "baselayer/MessageHandler";
+
+import * as API from "../API";
+import store from "../store";
+
+const REFRESH_INSTRUMENT = "skyportal/REFRESH_INSTRUMENT";
+
+const FETCH_INSTRUMENT = "skyportal/FETCH_INSTRUMENT";
+const FETCH_INSTRUMENT_OK = "skyportal/FETCH_INSTRUMENT_OK";
+
+const SUBMIT_INSTRUMENT = "skyportal/SUBMIT_INSTRUMENT";
+
+export const fetchInstrument = (id) =>
+  API.GET(`/api/instrument/${id}`, FETCH_INSTRUMENT);
+
+export const submitInstrument = (run) =>
+  API.POST(`/api/instrument`, SUBMIT_INSTRUMENT, run);
+
+// Websocket message handler
+messageHandler.add((actionType, payload, dispatch, getState) => {
+  const { instrument } = getState();
+  if (actionType === REFRESH_INSTRUMENT) {
```

Comment:
I think the reason the page doesn't update is that you need to send out a "REFRESH INSTRUMENT" on the back end (python handler for POST instrument) right before transmitting a success return statement.

Reply:
See below.